[Bugfix] crypto: dh - constify struct dh's pointer members

saeziae opened the pull request:
Const defined in `dh.h` but non-const in `dh.c`.

MichaIng commented on include/crypto/dh.h:
```diff
@@ -34,7 +34,6 @@
 struct dh {
 	const void *key;
 	const void *p;
-	const void *q;
```
Was this intended? The Q parameter is still used in `crypto/dh.c`, isn't it?

saeziae committed:
crypto: dh - constify struct dh's pointer members

struct dh contains several pointer members corresponding to DH parameters:
->key, ->p and ->g. A subsequent commit will introduce "dh" wrapping
templates of the form "ffdhe2048(dh)", "ffdhe3072(dh)" and so on in order
to provide built-in support for the well-known safe-prime ffdhe group
parameters specified in RFC 7919. These templates will need to set the
group parameter related members of the (serialized) struct dh instance
passed to the inner "dh" kpp_alg instance, i.e. ->p and ->g, to some
constant, static storage arrays.

Turn the struct dh pointer members' types into "pointer to const" in
preparation for this.

Signed-off-by: Nicolai Stange <nstange@suse.de>
Reviewed-by: Hannes Reinecke <hare@suse.de>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
